File: Book.js
class Book {
    // javascript does not support constructor overloading
    constructor() {
        if (arguments.length === 3) {
            // parameterized constructor
            this.title = arguments[0];
            this.author = arguments[1];
            this.pages = arguments[2];
        } else {
            // default constructor
            this.title = "No Title";
            this.author = "No Author";
            this.pages = 0;
        }
    }
}

// let book = new Book("Harry Potter", "J.K. Rowling", 500)
let book = new Book();

console.log(book);

File: Book.py
class Book:
    # default constructor is explicitly defined
    # python does not support constructor overloading

    # default constructor
    def __init__(self, *args):
        if len(args) == 3:
            # parametrized constructor
            self.title = args[0]
            self.author = args[1]
            self.pages = args[2]
        else:
            # default constructor
            self.title = ""
            self.author = ""
            self.pages = 0


# book1 = Book("Harry Potter", "JK Rowling", 500)
book1 = Book()

print(book1.title, book1.author, book1.pages)
